docs/DeveloperGuide.md

```diff
@@ -1,38 +1,269 @@
-# Developer Guide
+<div align="center">
+<h2> MoneyMind Developer Guide</h2>
+<h3> "Mind your Money" </h3>
+</div>
+
+<div align="left">
 
 ## Acknowledgements
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
+## Design
 
 {Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+### Architecture
+
+<img src="images\ArchitectureDiagram.png" width="300" />
+
+The ***Architecture Diagram*** given above explains the high-level design of the App. 
+
+Given below is a quick overview of how each component interact with each other.
+
+The main components in the architecture are:
+
+* `MoneyMind`: The main program of the application, 
+it initializes the other components in the correct sequence and is responsible for shut down the application.
+* `UI`: The user interface of the application.
+* `Storage`: The storage of data of the application.
+* `Data`: The data classes used in the application, including Event, Category and CategoryList.
+* `Command`: The command of the application, including AddCommand, DeleteCommand, ListCommand, etc.
+
+<img src="images\ArchitectureSequence.png" width="500" />
```

Comment:
I thought this sequence diagram is well done in conveying the idea 👍but there are a few notation mistakes. 
1. The entities should have an additional `:` to denote an unnamed instance of the classes.
2. The entities that are constructors should have its activation bar directly connected to the instance. e.g. `EventCommand` is constructed by `Parser`. 
3.  Instead of using `<<create>>` to signify the creation of an entity of the class, it is better to instantiate using `Class()`. An example will be from the textbook below:
![image](https://user-images.githubusercontent.com/88264296/228429094-1bdda42a-9afd-4a19-b9d2-8c28b52a1f05.png)
 
However, minor issue is the colour of the `Data` class makes it hard to see the return after a `Data` is created.